src/dataset-builder/ConceptCart.tsx

```diff
@@ -0,0 +1,26 @@
+import React, { Fragment } from 'react';
+import { ACTION_BAR_HEIGHT, ActionBar } from 'src/components/ActionBar';
+import { SnapshotBuilderConcept as Concept } from 'src/libs/ajax/DataRepo';
+
+interface ConceptCartProps {
+  cart: Concept[];
+  onClick: () => void;
+  actionText: string;
+}
+
+export const ConceptCart = (props: ConceptCartProps) => {
```

Comment:
It seems like slightly more refactoring could be done by refactoring the outer `Fragment` so the cart state could also be shared. But maybe doing that would require the outer object knowing too much about what the inner one looks like? What I'd like is a way to create a container that provides `cart` and `setCart` to its contents as bound objects. I guess that would require creating a type for that inner object and then implementing the two concept views as that type.. probably not worth the extra work.

Reply:
Yeah, I tend to prefer not to have the outer object knowing about the inner one. The components shouldn't really know about each other in my opinion.